person_module.rb

```diff
@@ -0,0 +1,32 @@
+require './student'
+require './teacher'
+
+module NewPeople
+  def create_student
+    input_obj = { 'Age: ' => '', 'Name: ' => '', 'Has Parent Permission: ' => '' }
+    input_obj.each do |key, _value|
+      print key
+      _value = gets.chomp
+    end
+    student = Student.new('Microverse', input_obj[0], input_obj[1])
+    @people << student
+    puts 'Person created successfully'
+  end
```

Comment:
- [ ] good job on cleaning up the `main.rb` file 👍 however this method is not according to the "single responsibility principle" all requests for input and input validation should've been placed in classes/methods whose responsibilities are to request or validate input. Please refer to my previous comment and [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for example and explanation.

Reply:
Thank you for your valuable comment, we solved it as you can see in the image below
![image](https://user-images.githubusercontent.com/60292761/141952156-d043814d-8844-4717-b6a8-a9b3dc4f5854.png)